mlrun/projects/project.py

```diff
@@ -1806,17 +1807,20 @@ def set_function(
             proj.set_function('my.py', requirements="requirements.txt")
 
         :param func:                function object or spec/code url, None refers to current Notebook
-        :param name:                name of the function (under the project), can be specified with a tag to support
-                                    versions (e.g. myfunc:v1)
+        :param name:                name of the function (under the project), can be specified with a tag that matches
+                                    the 'tag' parameter to support versions (e.g. myfunc:v1)
+                                    specifying a tag in the name will update the tagged function entry alone (myfunc:v1)
+                                    specifying a tag as a parameter will update the tagged function entry (myfunc:v1)
+                                    and the untagged entry (myfunc)
```

Comment:
I think this comment is pretty confusing, and users may not understand what it means. Also, since you're also describing the behavior in the `tag` parameter doc, I think it would be better to remove it from here and just keep the description there.